docs/content/development/extensions-core/sample-client-sslcontext.md

```diff
@@ -0,0 +1,27 @@
+---
+layout: doc_page
+---
+
+## Sample SSLContext Provider Module
+
+This module contains a sample implementation of [SSLContext](http://docs.oracle.com/javase/8/docs/api/javax/net/ssl/SSLContext.html)
+that will be injected to be used with HttpClient that Druid nodes use internally to communicate with each other. To learn more about
+Java's SSL support, please refer to [this](http://docs.oracle.com/javase/8/docs/technotes/guides/security/jsse/JSSERefGuide.html) guide.
+
+# Configuration
+
+|Property|Description|Default|Required|
+|--------|-----------|-------|--------|
+|`druid.client.https.algorithm`|SSL Algorithm/Protocol to use.|`null`|yes|
+|`druid.client.https.trustStoreType`|The type of the key store where trusted root certificates are stored.|`java.security.KeyStore.getDefaultType()`|yes|
+|`druid.client.https.trustStorePath`|The file path or URL of the TLS/SSL Key store where trusted root certificates are stored.|`null`|yes|
+|`druid.client.https.trustStoreAlgorithm`|Algorithm to be used by TrustManager to validate certificate chains|`javax.net.ssl.TrustManagerFactory.getDefaultAlgorithm()`|yes|
+
+All these configs must be set to some valid value. Apart from these properties, `trustStorePassword` is required to access trustStore.
+This password will be fetched using a `PasswordProvider` implementation.
+For example, `DefaultPasswordProvider` will look for `druid.password.trustStorePassword` runtime property to get these passwords.
+`EnvironmentVariablePasswordProvider` will directly look for value of `trustStorePassword` environment variable.
```

Comment:
Cool, the `DRUID_` + uppercasing convention for `EnvironmentVariablePasswordProvider` sounds good to me.